test/test_crud_v1.py

```diff
@@ -172,46 +172,9 @@ def run_scenario(self):
     return run_scenario
 
 
-def create_tests():
-    for dirpath, _, filenames in os.walk(_TEST_PATH):
-        dirname = os.path.split(dirpath)[-1]
-
-        for filename in filenames:
-            with open(os.path.join(dirpath, filename)) as scenario_stream:
-                scenario_def = json.load(scenario_stream)
-
-            test_type = os.path.splitext(filename)[0]
-
-            min_ver, max_ver = None, None
-            if 'minServerVersion' in scenario_def:
-                min_ver = tuple(
-                    int(elt) for
-                    elt in scenario_def['minServerVersion'].split('.'))
-            if 'maxServerVersion' in scenario_def:
-                max_ver = tuple(
-                    int(elt) for
-                    elt in scenario_def['maxServerVersion'].split('.'))
-
-            # Construct test from scenario.
-            for test in scenario_def['tests']:
-                new_test = create_test(scenario_def, test)
-                if min_ver is not None:
-                    new_test = client_context.require_version_min(*min_ver)(
-                        new_test)
-                if max_ver is not None:
-                    new_test = client_context.require_version_max(*max_ver)(
-                        new_test)
-
-                test_name = 'test_%s_%s_%s' % (
-                    dirname,
-                    test_type,
-                    str(test['description'].replace(" ", "_")))
-
-                new_test.__name__ = test_name
-                setattr(TestAllScenarios, new_test.__name__, new_test)
-
-
-create_tests()
+test_creator = TestCreator(create_test, TestAllScenarios, _TEST_PATH)
+test_creator.add_test_modifier(test_creator.enforce_min_max_server_version)
```

Comment:
As we discussed in person, I think we can call `enforce_min_max_server_version` on all these spec tests (transactions doesn't have the field but there's no harm in  supporting it there too). Then we can just remove `add_test_modifier` and the test specific skipping logic can happen in `create_test`.

Reply:
One issue I am running into while trying to implement this is the fact that some checks are relying upon the test name, which isn't readily available within `create_test` as things are currently written (we compute the test name after creating the test method and then `setattr` on the created method's `__name__`). 
I am modifying `create_test` to also accept the name as a parameter since that seems like the only sensible thing to do. Let me know if you have a better solution.

Reply:
You could also try using `self.id()`: https://docs.python.org/3/library/unittest.html#unittest.TestCase.id